File: LogicadeProgramacao/Estrutura de Decisão/3. Python/VerificaData.py
def data_valida(data):
    dia, mes, ano = map(int, data.split('/'))
    meses_com_31_dias = {1, 3, 5, 7, 8, 10, 12}
    meses_com_30_dias = {4, 6, 9, 11}
    
    if (ano % 4 == 0 and ano % 100 != 0) or ano % 400 == 0:
        fevereiro = 29
    else:
        fevereiro = 28
    
    if mes < 1 or mes > 12:
        return False
    
    if mes == 2:
        if dia < 1 or dia > fevereiro:
            return False
    elif mes in meses_com_31_dias:
        if dia < 1 or dia > 31:
            return False
    elif mes in meses_com_30_dias:
        if dia < 1 or dia > 30:
            return False
        return True

File: LogicadeProgramacao/Estrutura de Decisão/2. Java/VerificaData.java
import java.util.Scanner;

public class VerificaData {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Digite uma data no formato dd/mm/aaaa: ");
        String data = input.nextLine();

        int dia = Integer.parseInt(data.substring(0, 2));
        int mes = Integer.parseInt(data.substring(3, 5));
        int ano = Integer.parseInt(data.substring(6));

        boolean dataValida = true;

        if (ano < 1 || ano > 9999) {
            dataValida = false;
        } else if (mes < 1 || mes > 12) {
            dataValida = false;
        } else {
            switch (mes) {
                case 2:
                    if ((ano % 4 == 0 && ano % 100 != 0) || ano % 400 == 0) {
                        if (dia < 1 || dia > 29) {
                            dataValida = false;
                        }
                    } else {
                        if (dia < 1 || dia > 28) {
                            dataValida = false;
                        }
                    }
                    break;
                case 4:
                case 6:
                case 9:
                case 11:
                    if (dia < 1 || dia > 30) {
                        dataValida = false;
                    }
                    break;
                default:
                    if (dia < 1 || dia > 31) {
                        dataValida = false;
                    }
                    break;
            }
        }

        if (dataValida) {
            System.out.println("A data informada é válida.");
        } else {
            System.out.println("A data informada é inválida.");
        }

        input.close();
    }
}
